docs/current_docs/manuals/developer/module-structure.mdx

```diff
@@ -421,26 +386,33 @@ use (
 </TabItem>
 <TabItem value="Python">
 
-Dagger modules in Python are built to be installed, like libraries. The runtime container installs the module code with:
+Dagger modules in Python are built to be installed, like libraries. At module creation time, a `pyproject.toml` and `uv.lock` file will automatically be created that depend on the locally generated client library (in `./sdk`). This dependency is configured to be editable so that changes in the code don't require a re-install.
+
+With the `uv.lock` file, Dagger uses uv's [project management](https://docs.astral.sh/uv/concepts/projects/) capabilites, but it can be opted out by removing this file, in which case Dagger falls back to [the pip interface](https://docs.astral.sh/uv/pip/) instead.
+
+:::info
+Dagger also supports pinning dependencies with a `pip-tools` compatible **`requirements.lock`** file in order to support the use of other project managers like Poetry or Hatch locally (when developing).
+
+In this case, Dagger installs dependencies with:
 
 ```shell
 # executed by the runtime
-uv pip install -e .
+uv pip install -r requirements.lock -e ./sdk -e .
 ```
 
-Due to [PEP 517](https://peps.python.org/pep-0517/) and [`pip` supporting `pyproject.toml`](https://pip.pypa.io/en/stable/reference/build-system/pyproject-toml/), this means that all package managers that support the [build backend](https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#declaring-the-build-backend) standard are automatically supported.
+Notice that the `./sdk` and `main` (in `.`) packages don't need to be in the `requirements.lock` file, **only the third party dependencies**.
```

Comment:
Maybe this can be formatted as a note?

Reply:
I can't because it's already inside an "info" admonition and docusaurus doesn't render that well.